lib/misc.js

```diff
@@ -65,3 +65,51 @@ exports.generateId = function(fields) {
   return base32hex.encodeBuffer(hasher.digest(), { paddingChar: '-' });
 };
 
+var userIdsPath = function (env) {
+  return {
+    hashedPath: `${__dirname}/platform_users/${env}_hashed.json`,
+  };
+};
+
+var hashItem = function (item, salt) {
+  if (!item) {
+    throw new Error('Missing required userid');
+  }
+  if (!salt) {
+    throw new Error('Missing required salt');
+  }
+  return crypto.createHmac('sha256', salt).update(item).digest('hex');
+};
+
+exports.hashUserId = hashItem;
+
+// Function to encrypt the contents of a JSON file
+exports.addUserIdsToHashedEnvironmentFile = function (env, salt, unhashedUserIds) {
+  const filePaths = userIdsPath(env);
+
+  let hashedArray = [];
+  if (fs.existsSync(filePaths.hashedPath)) {
+    const data = fs.readFileSync(filePaths.hashedPath, 'utf8');
+    hashedArray = JSON.parse(data);
+  }
+
+  unhashedUserIds.forEach((userid) => {
+    const hashedId = hashItem(userid, salt);
+    if (!hashedArray.includes(hashedId)) {
+      hashedArray.push(hashItem(userid, salt));
+    }
+  });
+
+  fs.writeFileSync(filePaths.hashedPath, JSON.stringify(hashedArray));
+  console.log(`Hashed user IDs saved as ${filePaths.hashedPath}`);
+};
+
+exports.loadHashedPlatformUserId = function (env) {
+  const filePaths = userIdsPath(env);
+  if (!fs.existsSync(filePaths.hashedPath)) {
+    throw new Error(`Missing required file ${filePaths.hashedPath}`);
```

Comment:
Since this is going to fail if the hashed path for the environment doesn't exist, do we need to commit empty files (well, files with just `[]`) for every environment with this PR?